libs/zauth/src/Data/ZAuth/Token.hs

```diff
@@ -61,7 +71,16 @@ import Data.ByteString.Lazy.Char8 (split, break, drop)
 import Data.UUID
 import Sodium.Crypto.Sign (Signature (..))
 
-data Type = A | U | B | P deriving (Eq, Show)
+data Type = A -- ^ Access (Used as short-lived token for Users)
+          | U -- ^ User (Used as a cookie for Users to refresh access tokens)
+          | B -- ^ Bot
+          | P -- ^ Provider
+          | LA -- ^ LegalHold Access (Used as short-lived token for LegalHold Service)
+          | LU -- ^ LegalHold User (Used as a cookie for LegalHold Service to refresh access tokens)
+          deriving (Eq, Show)
+
+-- | Tag: Tokens for Users with no tag are refreshable themselves and called "UserToken"
+-- Tokens for Users with the tag 'S' are non-refreshable themselves and called "SessionToken"
```

Comment:
I would really like to rename this:

```haskell
data SessionTag = SessionTag
```

it's fine to leave the serialization intact, but at least the haskell code might as well readable.

(but not in this PR i guess; if you agree you can leave a `FUTUREWORK` comment.)